src/redux/categories/catogoriesSlice.js

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categoriesList: [],
+  status: '',
+};
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    add: (state, newCat) => {
+      // Redux Toolkit allows us to write "mutating" logic in reducers. It
+      // doesn't actually mutate the state because it uses the Immer library,
+      // which detects changes to a "draft state" and produces a brand new
+      // immutable state based off those changes
+      const list = state.categoriesList;
+      list.push(newCat);
+    },
+    checkStatus: (state) => {
+      const nState = state;
+      nState.status = 'Under construction';
+    },
+  },
```

Comment:
- [ ] The nState variable is assigned a reference to the original state object, which means any changes made to nState will also mutate the original state. This defeats the purpose of using immer and maintaining an immutable state. Example:
``
 reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
``